### PR TITLE
More documentation for idiomatic python API

### PR DESCRIPTION
**Pull request by alex-ozdemir**
Arithmetic, bit-vectors, arrays.

### CODE REVIEW

File: docs/api/python/z3compat/arith.rst [hunk 1]
```diff
@@ -1,2 +1,84 @@
...
+
+Arithmetic Operators
+-------------------
+.. autofunction:: cvc5_z3py_compat.ToReal
+.. autofunction:: cvc5_z3py_compat.ToInt
+.. autofunction:: cvc5_z3py_compat.IsInt
+.. autofunction:: cvc5_z3py_compat.Sqrt
+.. autofunction:: cvc5_z3py_compat.Cbrt
+
+See the following operator overloads for buildign other kinds of arithmetic
```
Comment (nafur) [inline docs/api/python/z3compat/arith.rst:31]:
```suggestion
See the following operator overloads for building other kinds of arithmetic
```

File: docs/api/python/z3compat/arith.rst [hunk 2]
```diff
@@ -1,2 +1,84 @@
...
+* :py:meth:`cvc5_z3py_compat.ArithRef.__mul__`
+* :py:meth:`cvc5_z3py_compat.ArithRef.__div__`
+* :py:meth:`cvc5_z3py_compat.ArithRef.__pow__`
+* :py:meth:`cvc5_z3py_compat.ArithRef.__neg__`
+* :py:meth:`cvc5_z3py_compat.ArithRef.__gt__`
+* :py:meth:`cvc5_z3py_compat.ArithRef.__lt__`
+* :py:meth:`cvc5_z3py_compat.ArithRef.__ge__`
+* :py:meth:`cvc5_z3py_compat.ArithRef.__le__`
+* :py:meth:`cvc5_z3py_compat.ExprRef.__eq__`
+* :py:meth:`cvc5_z3py_compat.ExprRef.__ne__`
```
Comment (nafur) [inline docs/api/python/z3compat/arith.rst:45]: Can we add the operation that is created, maybe as a definition list? (https://docutils.sourceforge.io/docs/ref/rst/restructuredtext.html#definition-lists)

File: docs/api/python/z3compat/array.rst [hunk 3]
```diff
@@ -1,2 +1,43 @@
...
+.. autofunction:: cvc5_z3py_compat.ArraySort
+
+Array Operators
+-------------------
+
+.. autofunction:: cvc5_z3py_compat.Update
+.. autofunction:: cvc5_z3py_compat.Store
+.. autofunction:: cvc5_z3py_compat.Select
+
+See the following operator overloads for buildign other kinds of array
```
Comment (nafur) [inline docs/api/python/z3compat/array.rst:19]:
```suggestion
See the following operator overloads for building other kinds of array
```

File: docs/api/python/z3compat/arith.rst [hunk 4]
```diff
@@ -1,2 +1,84 @@
...
+
+Testers
+-------------------
+.. autofunction:: cvc5_z3py_compat.is_arith_sort
```
Comment (nafur) [inline docs/api/python/z3compat/arith.rst:49]: This again looks a bit inconsistent. Why is there `is_arith_sort`, but not `is_real_sort`? Conversely, we have `is_int_value` but not `is_arith_value`. Why is `is_bool_value` here?

File: docs/api/python/z3compat/array.rst [hunk 5]
```diff
@@ -1,2 +1,43 @@
 Arrays
 ============
+
+
+Basic Array Term Builders
+-------------------------
+
+.. autofunction:: cvc5_z3py_compat.Array
+.. autofunction:: cvc5_z3py_compat.K
```
Comment (nafur) [inline docs/api/python/z3compat/array.rst:9]: Why is this called `K`? Do we have an alias for this with a reasonable name?

File: docs/api/python/z3compat/array.rst [hunk 6]
```diff
@@ -1,2 +1,43 @@
...
+-------------------
+
+.. autofunction:: cvc5_z3py_compat.Update
+.. autofunction:: cvc5_z3py_compat.Store
+.. autofunction:: cvc5_z3py_compat.Select
+
+See the following operator overloads for buildign other kinds of array
+terms:
+
+* select: :py:meth:`cvc5_z3py_compat.ArrayRef.__getitem__`
```
Comment (nafur) [inline docs/api/python/z3compat/array.rst:22]:
```suggestion
* ``select``: :py:meth:`cvc5_z3py_compat.ArrayRef.__getitem__`
```

File: docs/api/python/z3compat/array.rst [hunk 7]
```diff
@@ -1,2 +1,43 @@
...
+
+
+Testers
+-------------------
+.. autofunction:: cvc5_z3py_compat.is_array_sort
+.. autofunction:: cvc5_z3py_compat.is_array
+.. autofunction:: cvc5_z3py_compat.is_const_array
+.. autofunction:: cvc5_z3py_compat.is_K
+.. autofunction:: cvc5_z3py_compat.is_select
+.. autofunction:: cvc5_z3py_compat.is_store
```
Comment (nafur) [inline docs/api/python/z3compat/array.rst:32]: what about `is_update`?

File: docs/api/python/z3compat/z3compat.rst [hunk 8]
```diff
@@ -22,8 +22,3 @@ z3py compatibility Python API
     dt
```
Comment (nafur) [inline docs/api/python/z3compat/z3compat.rst:22]: Why is BV before arith and array?

File: docs/api/python/z3compat/bitvec.rst [hunk 9]
```diff
@@ -1,2 +1,81 @@
...
+.. autofunction:: cvc5_z3py_compat.SignExt
+.. autofunction:: cvc5_z3py_compat.ZeroExt
+.. autofunction:: cvc5_z3py_compat.RepeatBitVec
+.. autofunction:: cvc5_z3py_compat.BVRedAnd
+.. autofunction:: cvc5_z3py_compat.BVRedOr
+
+See the following operator overloads for buildign other kinds of bit-vector
+terms:
+
+* arithmetic
```
Comment (nafur) [inline docs/api/python/z3compat/bitvec.rst:38]: I would guess some BV people would prefer having explicit `BVAdd` and alike as well, to properly distinguish it from actual arithmetic expressions (and similarly for the Boolean operators).

File: docs/api/python/z3compat/boolean.rst [hunk 10]
```diff
@@ -2,16 +2,16 @@ Core & Booleans
 ================
 
 Basic Boolean Term Builders
--------------------
+-----------------------------
 .. autofunction:: cvc5_z3py_compat.Bool
-.. autofunction:: cvc5_z3py_compat.Bools
```
Comment (nafur) [inline docs/api/python/z3compat/boolean.rst:7]: Why is this removed?

File: docs/api/python/z3compat/array.rst (hunk 5 above)
Comment (alex-ozdemir) [reply docs/api/python/z3compat/array.rst:9]: I've added an alias.

File: docs/api/python/z3compat/array.rst (hunk 7 above)
Comment (alex-ozdemir) [reply docs/api/python/z3compat/array.rst:32]: I've added it.

File: docs/api/python/z3compat/z3compat.rst (hunk 8 above)
Comment (alex-ozdemir) [reply docs/api/python/z3compat/z3compat.rst:22]: I put the entries in order based on (a) a rough sense of importance and (b) the order in which I would want someone to read them in. Other options:

1. Keep this idea, but change the order, b/c, e.g., we this arith is more important to a new user.
2. Do the quickstart, followed by core, followed by alpha-order theories, followed by commands/internals/etc.
3. Do everything in alpha-order.

Comment (nafur) [reply docs/api/python/z3compat/z3compat.rst:22]: I'd favor 2, and I think it matches the "rough sense of importance":
1. Quickstart
2. Core / Boolean
3. Theories
4. Internals

Comment (alex-ozdemir) [reply docs/api/python/z3compat/z3compat.rst]: Sounds good.

File: docs/api/python/z3compat/arith.rst (hunk 4 above)
Comment (alex-ozdemir) [reply docs/api/python/z3compat/arith.rst]: I moved `is_bool_value`. Good catch!

It seems that these testing functions are designed to line up with the classes. I'm not really sure what to do with them. To be frank, I do not think they're very useful, but we can't remove the existing ones without breaking with Z3Py.

File: docs/api/python/z3compat/dummy.rst [hunk 11]
```diff
@@ -0,0 +1,27 @@
+Bit-Vectors
```
Comment (nafur) [inline docs/api/python/z3compat/dummy.rst:1]: Did you want to commit this?

**Review by nafur — APPROVED** [review]
Just one thing about the `dummy` file, otherwise LGTM.

File: docs/api/python/z3compat/dummy.rst (hunk 11 above)
Comment (alex-ozdemir) [reply docs/api/python/z3compat/dummy.rst]: Whoops; good catch. Removed.
